src/commonMain/kotlin/io/github/koalaplot/core/xygraph/XYGraph.kt

```diff
@@ -172,18 +217,32 @@ public fun <X, Y> XYGraph(
     }
 }
 
-private fun <T> transformAxis(
+private fun <T> zoomAxis(
     axis: AxisModel<T>,
     length: Int,
     centroid: Float,
-    pan: Float,
     zoom: Float
 ) {
     val pivot = (length.toFloat() - centroid) / length.toFloat()
     axis.zoom(zoom, pivot.coerceIn(0f, 1f))
+}
+
+private fun <T> panAxis(
+    axis: AxisModel<T>,
+    length: Int,
+    pan: Float
+) {
     axis.pan(pan / length.toFloat())
 }
 
+private fun Offset.normalizeCentroid(size: IntSize): Offset = copy(
+    x = abs(x - size.width),
```

Comment:
why doesn't the y-axis need to be adjusted here as well? Would be worth adding comment for future reference.

Reply:
There was a bug that I discovered back in the previous implementation. It was associated with incorrect zoom application due to different coordinate reference points. In fact, there should be normalization along the Y axis and the zoom method can be simplified to avoid unnecessary subtraction.
Added a comment, thanks :)